### ISSUE TITLE
[Feature]: be able to use iframes without padding

### ISSUE THREAD

**Issue by toxicflux** [appsmithorg/appsmith#35555]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Iframes are currently useless for small components because padding is added and is not related to external component.
This makes iframes since it is not possible to align smaller iframes as close as I want.  Making height smaller adds an annoying scroll bar on overflow.

![image](https://github.com/user-attachments/assets/1e031aaa-dc64-4a99-80e6-f0a377118cc7)

I want to have zero padding for iframe or possibility to remove it.

### Why should this be worked on?

We want to replace small components with components of our own design, hosted externally.
Iframes works for larger widgets, but is useless for smaller widgets because of padding (left, right, top and bottom).

**Comment by Nikhil-Nandagopal** [comment]
@toxicflux have you tried using the custom widget for this?

**Comment by toxicflux** [comment]
Yes, I've tried Custom widget for this (as screenshot show). Custom widget do not work for reusable code, it has severe performance issues  and it doesn't work putting an iframe inside a custom widget as I described above.

**Comment by Nikhil-Nandagopal** [comment]
Ok I'm not sure iFrame is the best idea either because many iFrames on a single page will likely lead to a performance issue as well. Could you tell us why you are looking to customise the components? We are working on reusable widgets as well but it doesn't seem like that will suit your use case either.

**Comment by toxicflux** [comment]
We need to style our applications to meet customer requirements, meaning we want to override the style of basic components such as buttons, inputs, selects. The styling of appsmith components are limited, the best would be if we could provide our own component library.

**Comment by carinanfonseca** [comment]
@toxicflux, thank you for raising this topic. We’d like to understand more about the use cases and limitations, which will help us identify what's missing in the Appsmith product and potential improvements.

Please feel free to [book a time](https://calendly.com/carina-neves-fonseca/appsmith-feedback) that works for you. I look forward to speaking with you.

**Comment by nikguhl89** [comment]
This is also something I have had issues with. I am trying to display an external website within an iframe, but it is zoomed in too much to fully fit the page in view. This is especially a problem on the mobile view. If I had the ability to fit content to width on the iframe at least this would be much more usable. I tried doing a custom iframe, but I was not able to interact with my content, only display it.
